### ISSUE TITLE
Exception retrieving extension handlers: local variable 'goal_state' referenced before assignment

### ISSUE THREAD

**Issue by boumenot** [Azure/WALinuxAgent#1026]
I discovered this in 2.2.21.

```
Exception retrieving extension handlers: local variable 'goal_state' referenced before assignment  Exception retrieving extension handlers: [ProtocolError] Exceeded max retry updating goal state
```

It is unclear if this issue has existed previous to 2.2.21, or if it is new.  Based on my review it appears to have existed previous to this version.

**Comment by hglkrijger** [comment]
Changes merged for more detailed telemetry. When that starts flowing we can make further progress on the underlying issue.

**Comment by vrdmr** [comment]
In the past 30 days - This only showed in two versions -

```
Version	dcount_ContainerId
2.2.20	1018
2.2.19	952
```

**Comment by vrdmr** [comment]
Most of the goal_state variable name issue was resolved by #1392 and now the new ones as seen in the newer agents are seen in [get_ext_conf()](https://github.com/Azure/WALinuxAgent/blob/develop/azurelinuxagent/common/protocol/wire.py#L892)

Example messages from Telemetry:
```
Exception retrieving extension handlers: local variable 'goal_state' referenced before assignment 
[
<FrameSummary file /usr/lib/python3/dist-packages/azurelinuxagent/ga/exthandlers.py, line 219 in run>, 
<FrameSummary file /usr/lib/python3/dist-packages/azurelinuxagent/common/protocol/wire.py, line 153 in get_ext_handlers>,
<FrameSummary file /usr/lib/python3/dist-packages/azurelinuxagent/common/protocol/wire.py, line 894 in get_ext_conf>
]
```

```
Exception processing extension handlers: local variable 'goal_state' referenced before assignment [
 <FrameSummary file /usr/lib/python3/dist-packages/azurelinuxagent/ga/exthandlers.py, line 253 in run>,
 <FrameSummary file /usr/lib/python3/dist-packages/azurelinuxagent/ga/exthandlers.py, line 514 in report_ext_handlers_status>,
 <FrameSummary file /usr/lib/python3/dist-packages/azurelinuxagent/common/protocol/wire.py, line 192 in report_vm_status>,
 <FrameSummary file /usr/lib/python3/dist-packages/azurelinuxagent/common/protocol/wire.py, line 1018 in upload_status_blob>,
 <FrameSummary file /usr/lib/python3/dist-packages/azurelinuxagent/common/protocol/wire.py, line 894 in get_ext_conf>]
```

**Comment by vrdmr** [comment]
Looks like #1691 seems to solve the issue. No new occurrence in 2.2.46 version. Closing the issue. If it returns, will reopen the issue.
